lib/puppet/provider/templates/invoke_dsc_resource.ps1.erb

```diff
@@ -43,6 +43,6 @@ switch ($invokeParams.Method) {
     [bool]($result.InDesiredState)
   }
   'Set' {
-    @($Null, 0) -contains $result.ReturnValue
+    [bool]($result.RebootRequired)
```

Comment:
This seems like a peculiar value to return as the succeeds / fails for the `Set` operation.

As I understand it, if `Invoke-DscResource` didn't throw an exception, then the operation succeeded.

Since we don't have the ability to (yet) accept a more structured response in Ruby, I wonder if we're best off just returning `true` here for now.

Otherwise, we have to inspect exit code for pass / fail state, then realize that a `true` means succeeded + reboot required and `false` means succeeded + no reboot required.  Those are some weird semantics.

Alternatively, I think that it would make sense to do the exception catching / structured JSON output now - at least for the results of these operations.  I think that's likely going to take care of  [MODULES-2194](https://tickets.puppetlabs.com/browse/MODULES-2194)  as well which is on this sprints list of things to do.